core/modules/replica/ReplicationRequest.cc

```diff
@@ -0,0 +1,313 @@
+                                    int  priority,
+                                    bool keepTracking,
+                                    bool allowDuplicate,
+                                    std::shared_ptr<Messenger> const& messenger)
+    :   RequestMessenger(serviceProvider,
+                         io_service,
+                         "REPLICA_CREATE",
+                         worker,
+                         priority,
+                         keepTracking,
+                         allowDuplicate,
+                         messenger),
+        _database(database),
+        _chunk(chunk),
+        _sourceWorker(sourceWorker),
+        _onFinish(onFinish),
+        _replicaInfo() {
+
+    _serviceProvider->assertWorkerIsValid(sourceWorker);
+    _serviceProvider->assertWorkersAreDifferent(sourceWorker, worker);
+    _serviceProvider->assertDatabaseIsValid(database);
+}
+
+void ReplicationRequestM::startImpl() {
+
+    LOGS(_log, LOG_LVL_DEBUG, context() << "startImpl");
+
+    // Serialize the Request message header and the request itself into
+    // the network buffer.
+
+    _bufferPtr->resize();
+
+    proto::ReplicationRequestHeader hdr;
+    hdr.set_id(id());
+    hdr.set_type(proto::ReplicationRequestHeader::REPLICA);
+    hdr.set_replica_type(proto::ReplicationReplicaRequestType::REPLICA_CREATE);
+
+    _bufferPtr->serialize(hdr);
+
+    proto::ReplicationRequestReplicate message;
+    message.set_priority(priority());
+    message.set_database(database());
+    message.set_chunk(chunk());
+    message.set_worker(sourceWorker());
+
+    _bufferPtr->serialize(message);
+
+    send();
+}
+
+void ReplicationRequestM::wait() {
+
+    LOGS(_log, LOG_LVL_DEBUG, context() << "wait");
+
+    // Allways need to set the interval before launching the timer.
+
+    _timer.expires_from_now(boost::posix_time::seconds(_timerIvalSec));
+    _timer.async_wait(
+        boost::bind(
+            &ReplicationRequestM::awaken,
+            shared_from_base<ReplicationRequestM>(),
+            boost::asio::placeholders::error
+        )
+    );
+}
+
+void ReplicationRequestM::awaken(boost::system::error_code const& ec) {
+
+    LOGS(_log, LOG_LVL_DEBUG, context() << "awaken");
+
+    if (isAborted(ec)) { return; }
+
+    // Also ignore this event if the request expired
+    if (_state== State::FINISHED) { return; }
+
+    LOCK_GUARD;
+
+    // Serialize the Status message header and the request itself into
+    // the network buffer.
+
+    _bufferPtr->resize();
+
+    proto::ReplicationRequestHeader hdr;
+    hdr.set_id(id());
+    hdr.set_type(proto::ReplicationRequestHeader::REQUEST);
+    hdr.set_management_type(proto::ReplicationManagementRequestType::REQUEST_STATUS);
+
+    _bufferPtr->serialize(hdr);
+
+    proto::ReplicationRequestStatus message;
+    message.set_id(remoteId());
+    message.set_type(proto::ReplicationReplicaRequestType::REPLICA_CREATE);
+
+    _bufferPtr->serialize(message);
+
+    send();

```

Comment:
Holding a lock while you're sending messages out on the wire seems to me like asking for trouble. I don't know the larger context well enough to say this is wrong, but I will say it raises a red flag for me and maybe there's a better way?